include/zephyr/drivers/gpio.h

```diff
@@ -129,6 +129,14 @@ extern "C" {
  */
 #define GPIO_INT_HIGH_1                (1U << 26)
 
+/* Disable/Enable interrupt functionality without changing other interrupt
+ * related register, such as clearing the pending register.
+ *
+ * This is a component flag that should be combined with `GPIO_INT_ENABLE` or
+ * `GPIO_INT_DISABLE` flags to produce a meaningful configuration.
+ */
+#define GPIO_INT_ENABLE_DISABLE_ONLY   (1u << 27)
```

Comment:
This should be guarded with CONFIG_GPIO_ENABLE_DISABLE_INTERRUPT.  So that applications will get a compile error if trying to use this bit without the config enabled.

Reply:
Done